ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/src/JacksonConverter.kt

```diff
@@ -95,6 +121,59 @@ public class JacksonConverter(
             }
         }
     }
+
+    private companion object {
+        private const val beginArrayCharCode = '['.code
+        private const val endArrayCharCode = ']'.code
+        private const val objectSeparator = ','.code
+    }
+
+    /**
+     * Guaranteed to be called inside a [Dispatchers.IO] context, see [OutputStreamContent]
+     */
+    @Suppress("BlockingMethodInNonBlockingContext")
+    private suspend fun <T> Flow<T>.serializeJson(outputStream: OutputStream) {
+        val jfactory = JsonFactory()
+        // cannot use ObjectMapper write to Stream because it flushes the OutputStream on each write
+        val jGenerator = jfactory.createGenerator(outputStream, JsonEncoding.UTF8)
+        jGenerator.prettyPrinter = MinimalPrettyPrinter("") // avoid single space between items
+        jGenerator.codec = objectMapper
+
+        outputStream.write(beginArrayCharCode)
+        collectIndexed { index, value ->
+            if (index > 0) {
+                outputStream.write(objectSeparator)
+                outputStream.flush()
+            }
+            jGenerator.writeObject(value)
+        }
+        outputStream.write(endArrayCharCode)
+    }
+
+    /**
+     * Guaranteed to be called inside a [Dispatchers.IO] context, see [OutputStreamContent]
+     */
+    @Suppress("BlockingMethodInNonBlockingContext")
+    private suspend fun <T> Flow<T>.serializeJson(writer: Writer) {
+        val jfactory = JsonFactory()
+        // cannot use ObjectMapper write to Stream because it flushes the OutputStream on each write
+        val jGenerator = jfactory.createGenerator(writer)
+        jGenerator.prettyPrinter = MinimalPrettyPrinter("") // avoid single space between items
+        jGenerator.configure(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM, false)
```

Comment:
Because the normal behavior is to flush every time a Jackson `write*` that targets the IO Writer is called (= the value in the Flow), that's not what I want, I want to flush only after the separator is written.